### PR TITLE
[WIP] Implement AsGraph wrapper for dataio iterators

### PR DESCRIPTION
**Pull request by Drew-Wagner**
# WORK IN PROGRESS
- [x] Implement AsGraph wrapper to convert regular data loader to torch_geometric
- [x] Integrate with train.py

### CODE REVIEW

**Comment by Drew-Wagner** [comment]
@vmcru If this is useful for you, then I can make a separate PR with just he last 3 commits so we can make a PR in SB-MOABB

File: benchmarks/MOABB/utils/dataio_iterators.py [hunk 1]
```diff
@@ -157,12 +114,8 @@ def get_neighbour_channels(
     return sel_channels
 
 
-def sample_channels(x, adjacency_mtx, ch_names, n_steps, seed_nodes=["Cz"], keep_all_channels=False):
+def sample_channels(x, adjacency_mtx, ch_names, n_steps, seed_nodes=["Cz"]):
```
Comment (engmubarak48) [inline benchmarks/MOABB/utils/dataio_iterators.py:117]: somehow the ch_names are less than the len of the adjacency_mtx leading issues when we call `get_neighbour_channels`

File: benchmarks/MOABB/utils/dataio_iterators.py [hunk 2]
```diff
@@ -424,6 +404,80 @@ def get_tail_path(self, subject: int, session: Optional[str]) -> str:
...
+            save_prepared_dataset=save_prepared_dataset,
+        )
+
+        edge_list = adjacency_mtx_to_edge_list(datasets["adjacency_mtx"])
+
+        for key in ("train", "valid", "test"):
+            datasets[key] = self.make_graph_dataloader(
+                datasets[key],
+                edge_list=edge_list,
+                ch_positions=torch.from_numpy(datasets["ch_positions"]),
```
Comment (engmubarak48) [inline benchmarks/MOABB/utils/dataio_iterators.py:455]: I think this should be `channel_positions` or changed above

**Review by engmubarak48 — COMMENTED** [review]
Good, few fixes needed

File: benchmarks/MOABB/utils/dataio_iterators.py (hunk 1 above)
Comment (engmubarak48) [reply benchmarks/MOABB/utils/dataio_iterators.py:117]: this is an issue when we use the `n_steps_channel_selection: 2`

File: benchmarks/MOABB/utils/dataio_iterators.py [hunk 3]
```diff
@@ -424,6 +402,80 @@ def get_tail_path(self, subject: int, session: Optional[str]) -> str:
...
+        self,
+        dataloader: DataLoader,
+        edge_list: torch.Tensor,
+        ch_positions: torch.Tensor,
+        shuffle: bool = False,
+    ) -> torch_geometric.loader.DataLoader:
+        dataset = [
+            torch_geometric.data.Data(x=x, y=y, edge_index=edge_list)
+            for x, y in dataloader.dataset
+        ]
```
Comment (engmubarak48) [inline benchmarks/MOABB/utils/dataio_iterators.py]: @Drew-Wagner I think for the graph features shall we reshape the data here and have it like this as x: (C, T). We can also do this in the forward if you think is more better :)

**Comment by engmubarak48** [comment]
@Drew-Wagner this looks good to me.. we can merge if you are okay on your side?

**Comment by Drew-Wagner** [comment]
Looks good to me!

**Pull request merged**